eos/maths/gegenbauer-polynomial.hh

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 Danny van Dyk
+ * Copyright (c) 2022 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_MATHS_GEGENBAUER_POLYNOMIALS_HH
+#define EOS_GUARD_EOS_MATHS_GEGENBAUER_POLYNOMIALS_HH 1
+
+#include <eos/maths/power-of.hh>
+
+#include <vector>
+
+namespace eos
+{
+    class GegenbauerPolynomial
+    {
+        private:
+            const unsigned int order;
+
+            const double alpha; // alpha > -0.5
+
+            std::vector<double> coefficients;
+
+            const unsigned int halforder;
+
+            std::vector<double> calculate_coefficients() const
+            {
+                const unsigned int isodd = order % 2;
```

Comment:
Would be helpful to store this in the class object itself upon construction. Maybe a member variable ``order_is_odd``, which is boolean?

Reply:
I don't understand why that would be that helpful, but I can do that. however I need it as integer since it appears in the calculation of the coefficients

Reply:
In that case, name it to something that is more obvious to the reader, e.g., ``power_shift`` and set it explicitly to an integer value at the construction time of the polynomial:
```C++
const int power_shift = (1 == order % 2) ? 1 : 0; // we need to shift the power in some expressions by 1 if the order is odd
```

Reply:
I implemented the factor as suggested in the first message, as it appears also in the evaluate method